source/client/HandManagement.py

```diff
@@ -0,0 +1,114 @@
+                card_wrapped = UICardWrapper(card, card_xy, self.hand_scaling[0])
+            updated_wrapped_hand.append(card_wrapped)
+        maxX = UIC.Disp_Width - (self.hand_scaling[1] / 2)
+        if (card_xy[0] > maxX):
+            scalingfactor = maxX / card_xy[0]
+            self.hand_scaling = (scalingfactor * self.hand_scaling[0], scalingfactor * self.hand_scaling[1])
+            updated_wrapped_hand = rescaleCards(self, updated_wrapped_hand, self.hand_scaling[0])
+            self.refresh_flag = True
+        if (self.hand_scaling[0] != UIC.scale and len(updated_wrapped_hand) <= self.deal_size):
+            self.hand_scaling = (UIC.scale, UIC.Card_Spacing)
+            updated_wrapped_hand = rescaleCards(self, updated_wrapped_hand, self.hand_scaling[0])
+        return updated_wrapped_hand
+
+def clearPreparedCardsGui(self):
+    for element in self.hand_info:
+        if element.status == 2:
+            element.status = 0
+            element.img_clickable.changeOutline(0)
+
+def preparedCardsPlayedGui(self):
+    if len(self.controller.prepared_cards) == 0:
+        # manually remove cards played, else an ambiguity in wrapped cards causes
+        # picked up cards to sometimes get coordinates of cards just played.
+        # If statement insures that you don't remove cards if pick-up failed.
+        for wrappedcard in self.hand_info:
+            if wrappedcard.status == 2:
+                self.hand_info.remove(wrappedcard)
+                self.last_hand.remove(wrappedcard.card)
+
+
+def refreshXY(self, original, layout_option=1):
+    self.refresh_flag = False
+    """After sorting or melding, may wish to refresh card's xy coordinates """
+    maxX = UIC.Disp_Width - (self.hand_scaling[1] / 2)
+    if not layout_option == 1:
+        print('the only layout supported now is cards in a line, left to right')
+    refreshed = []
+    card_xy = (10, UIC.Table_Hand_Border + 40)
+    for element in original:
+        element.img_clickable.x = card_xy[0]
+        element.img_clickable.y = card_xy[1]
+        card_xy = (card_xy[0] + self.hand_scaling[1], card_xy[1])
+        refreshed.append(element)
+    if (card_xy[0] > maxX):
+        scalingfactor = maxX / card_xy[0]
+        self.hand_scaling = (scalingfactor * self.hand_scaling[0], scalingfactor * self.hand_scaling[1])
+        refreshed = rescaleCards(refreshed, self.hand_scaling[0])
+    return refreshed
+
+def rescaleCards(self, original, card_scaling):
+    rescaled = []
+    for element in original:
+        loc_xy = (element.img_clickable.x, element.img_clickable.y)
+        scaledelement = UICardWrapper(element.card, loc_xy, card_scaling)
+        scaledelement.status = element.status
+        scaledelement.img_clickable.outline_index = element.img_clickable.outline_index
+        rescaled.append(scaledelement)
+    self.refresh_flag = True
+    return rescaled
+
+def showHolding(self, wrapped_cards):
```

Comment:
Maybe call this DrawWrappedHand or DrawHand or something with render? I would ideally choose something similar to what is being used for the name of the button render method.

Reply:
I like showHolding.